### PR TITLE
Adding end to end global.json SDK resolution test

### PR DESCRIPTION
**Pull request by sfoslund**
Retargeting https://github.com/dotnet/sdk/pull/13211

### CODE REVIEW

**Comment by sfoslund** [comment]
@dsplaisted this is bizarre, tests are consistently passing when I trigger CI manually, but they failed when I initially made this PR. There shouldn't be any difference based on the way the build is triggered, right?

**Comment by dsplaisted** [comment]
> @dsplaisted this is bizarre, tests are consistently passing when I trigger CI manually, but they failed when I initially made this PR. There shouldn't be any difference based on the way the build is triggered, right?

Not AFAIK.  @wli3 @rainersigwald @mmitche @MattGal

**Comment by MattGal** [comment]
> > @dsplaisted this is bizarre, tests are consistently passing when I trigger CI manually, but they failed when I initially made this PR. There shouldn't be any difference based on the way the build is triggered, right?
> 
> Not AFAIK. @wli3 @rainersigwald @mmitche @MattGal

Took a peek, this isn't likely our infra, rather it's either the desktop msbuild in use for these tests or the SDK resolving task just not correctly logging errors.  I can help @sfoslund with investigation machines that match a given build as needed though.

**Comment by sfoslund** [comment]
Well, apparently it doesn't repro anymore, even on PR triggered runs (see https://github.com/dotnet/sdk/pull/14131). @MattGal it seems like you were right, it could have had something to do with the preview builds.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildWithGlobalJson.cs [hunk 1]
```diff
@@ -16,29 +18,60 @@ public class GivenThatWeWantToBuildWithGlobalJson : SdkTest
...
-        public void It_fails_build_on_failed_sdk_resolution()
+        [FullMSBuildOnlyTheory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void It_fails_build_on_failed_sdk_resolution(bool runningInVS)
         {
-            var fakePath = "fakePath";
-            TestProject testProject = new TestProject()
+            var prevIncludeDefault = Environment.GetEnvironmentVariable("MSBUILDINCLUDEDEFAULTSDKRESOLVER");
+            var prevAdditionalFolder = Environment.GetEnvironmentVariable("MSBUILDADDITIONALSDKRESOLVERSFOLDER");
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildWithGlobalJson.cs:27]: What about setting up these environment variables for all tests (when running on full framework)?  That was flaky last time we tried it, but I'd still like to try to turn that back on.  It doesn't necessarily have to be in this PR though.

**Review by dsplaisted — APPROVED** [review]

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildWithGlobalJson.cs (hunk 1 above)
Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildWithGlobalJson.cs:27]: Actually, it looks like we're already setting this up: https://github.com/dotnet/sdk/blob/release/5.0.1xx/src/Tests/Microsoft.NET.TestFramework/ToolsetInfo.cs#L154

So it seems like setting this variable here may be redundant.

**Pull request merged**